docs/specs/event-scanner.md

```diff
@@ -0,0 +1,383 @@
+}
+
+interface BlockNumbers {
+  [date: string]: BlockNumberEntry; // YYYY-MM-DD format
+}
+```
+
+### Output Types
+
+```typescript
+interface OutflowEvent {
+  recipient: string; // Checksummed Ethereum address
+  value_wei: string; // Decimal string representation
+  tx_hash: string; // 0x-prefixed transaction hash
+}
+
+interface DailyOutflow {
+  block_number: number; // Last block of the day
+  total_outflow_wei: string; // Sum of all event values
+  events: OutflowEvent[]; // All events for the day
+}
+
+interface OutflowData {
+  metadata: {
+    chain_id: number; // Network chain ID
+    reward_distributor: string; // Distributor contract address
+  };
+  outflows: {
+    [date: string]: DailyOutflow; // YYYY-MM-DD format
+  };
+}
+```
+
+### Event Structure
+
+```typescript
+// RecipientRecieved event (note intentional misspelling)
+// event RecipientRecieved(address indexed recipient, uint256 value)
+interface RecipientRecievedEvent {
+  topics: [string, string]; // [eventSignature, recipientAddress]
+  data: string; // Encoded uint256 value
+  transactionHash: string; // Transaction that emitted the event
+  address: string; // Contract that emitted the event
+}
+```
+
+## 8. Implementation Requirements
+
+### Validation Rules
+
+1. **Input Validation**
+
+   - Distributor address must be valid Ethereum address
+   - Distributor must exist in distributors.json
+   - Block numbers file must contain required date ranges
+   - Dates must be in YYYY-MM-DD format
+
+2. **Event Validation**
+
+   - Event must be from specified distributor contract
+   - Recipient must be valid Ethereum address
+   - Value must be valid uint256 (non-negative, within bounds)
+   - Transaction hash must be 66 characters (0x + 64 hex)
+
+3. **Output Validation**
+   - Total outflow must equal sum of event values
+   - All monetary values stored as decimal strings
+   - Addresses stored in checksummed format
+   - No duplicate events within a day
+
+### Security Considerations
+
+1. **Data Integrity**
+
+   - Validate all blockchain data before processing
+   - Use checksummed addresses for consistency
+   - Atomic file updates (write to temp, then rename)
+
+2. **Error Handling**
+
+   - Never partially update outflow data
+   - Preserve existing data on failure
+   - Clear error messages with context
+
+3. **Resource Management**
+   - Process events in manageable batches
+   - Limit concurrent RPC requests
+   - Clean up resources on failure
+
+## 9. Test Data Requirements
+
+### Minimum Test Data
+
+1. At least one deployed reward distributor contract
+2. Historical `RecipientRecieved` events spanning multiple days
+3. Master block numbers file covering all test dates
+4. Various event patterns:
+   - Single event days
+   - Multiple event days
+   - Days with no events
+   - Large value transfers
+   - Small value transfers
+
+### Data Sources
+
+- Mainnet Arbitrum reward distributor contracts
+- Historical blockchain data via archive node
```

Comment:
dont need archive for this - and there is an rpc in other tests that we'll use the same as